From f7b659b0d439bb40e104a008c2375fa79b60fb5e Mon Sep 17 00:00:00 2001
From: Owen Huang <Owen.Huang@infineon.com>
Date: Mon, 27 Mar 2023 01:47:54 -0500
Subject: [PATCH 182/208] brcmfmac: move SDIO function number definition to
 sdio.h

1. adding include/linux/mmc/card.h into backport list will cause sdio probe
failed


Signed-off-by: Owen Huang <Owen.Huang@infineon.com>
---
 .../net/wireless/broadcom/brcm80211/brcmfmac/sdio.h    | 10 ++++++++++
 include/linux/mmc/card.h                               |  9 ---------
 2 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.h b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.h
index d4166805a197..957eb36d3771 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.h
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.h
@@ -147,6 +147,16 @@
 /* watchdog polling interval */
 #define BRCMF_WD_POLL	msecs_to_jiffies(10)
 
+/* SDIO function number definition */
+#define SDIO_FUNC_0		0
+#define SDIO_FUNC_1		1
+#define SDIO_FUNC_2		2
+#define SDIO_FUNC_3		3
+#define SDIO_FUNC_4		4
+#define SDIO_FUNC_5		5
+#define SDIO_FUNC_6		6
+#define SDIO_FUNC_7		7
+
 /**
  * enum brcmf_sdiod_state - the state of the bus.
  *
diff --git a/include/linux/mmc/card.h b/include/linux/mmc/card.h
index 2b6d8bffdb16..37f975875102 100644
--- a/include/linux/mmc/card.h
+++ b/include/linux/mmc/card.h
@@ -236,15 +236,6 @@ struct mmc_queue_req;
 
 #define SDIO_MAX_FUNCS		7
 
-#define SDIO_FUNC_0		0
-#define SDIO_FUNC_1		1
-#define SDIO_FUNC_2		2
-#define SDIO_FUNC_3		3
-#define SDIO_FUNC_4		4
-#define SDIO_FUNC_5		5
-#define SDIO_FUNC_6		6
-#define SDIO_FUNC_7		7
-
 /* The number of MMC physical partitions.  These consist of:
  * boot partitions (2), general purpose partitions (4) and
  * RPMB partition (1) in MMC v4.4.
-- 
2.17.1

